lib/graphiti/resource/remote.rb

```diff
@@ -0,0 +1,48 @@
+module Graphiti
+  class Resource
+    module Remote
+      extend ActiveSupport::Concern
+
+      included do
+        self.adapter = Graphiti::Adapters::JSONAPI
+        self.model = OpenStruct
+        self.validate_endpoints = false
+
+        class_attribute :timeout,
+          :open_timeout
+      end
+
+      def save(*args)
+        raise Errors::RemoteWrite.new(self.class)
+      end
+
+      def destroy(*args)
+        raise Errors::RemoteWrite.new(self.class)
+      end
+
+      def before_resolve(scope, query)
+        scope[:params] = Util::RemoteParams.generate(self, query)
+        scope
+      end
+
+      # Forward all headers
+      def request_headers
+        if defined?(Rails)
+          context.request.headers.to_h.reject { |k, v| k.include?('.') }
+        else
+          {}
+        end
+      end
+
+      def make_request(url)
+        headers = request_headers.dup
+        headers['Content-Type'] = 'application/vnd.api+json'
+        Faraday.get(url, nil, headers) do |req|
+          yield req if block_given? # for super do ... end
+          req.options.timeout = timeout if timeout
```

Comment:
Instead of adding a bunch of faraday-specific class attributes and moving them over (and dealing with the PRs that will come in asking for new ones), I'd rather see something like `def faraday_request_options(options)` which lets users modify whatever they need.

Reply:
I actually see `timeout` and `open_timeout` as not specific to Faraday. The underlying `Net::HTTP` implementation actually uses these values, so I view it as a general purpose thing.

Since faraday is geared more towards middleware, I think customizing faraday with a block instead of a hash is more appropriate:

```ruby
def make_request(url)
  super do |req|
    # customize faraday request
  end
end
```

Reply:
Ahh I had actually forgotten `open_timeout` worked on underlying `Net::HTTP` as well. I'm fine with this then if we aren't planning on extending this option set any further.